### PR TITLE
Fix ci/build.py using remote cache locally

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
Minor PR to fix ci/build.py rebuilding local cache all the time

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:

### CODE REVIEW

File: ci/build.py [hunk 1]
```diff
@@ -221,56 +221,62 @@ def script_name() -> str:
...
                         action='store_true')
 
-    parser.add_argument("--docker-registry",
-                        help="Dockerhub registry name to retrieve cache from",
+    parser.add_argument("-d", "--docker-registry",
+                        help="Dockerhub registry name to retrieve cache from. Default is 'mxnetci'",
                         default='mxnetci',
                         type=str)
 
+    parser.add_argument("-c", "--cache", action="store_true",
```
Comment (marcoabreu) [inline ci/build.py]: We already have that check in https://github.com/larroy/mxnet/blob/1aca0a1d3fe772f13a957175ddb42449c393f585/ci/build.py#L164. Just remove the default value in https://github.com/larroy/mxnet/blob/1aca0a1d3fe772f13a957175ddb42449c393f585/ci/build.py#L226 and we're good.

I don't see much reason to have two options to manage this functionality

Comment (marcoabreu) [reply ci/build.py]: Please don't forget that we actually WANT the people to use the cache locally. If there's a problem, we should fix it.

Comment (larroy) [reply ci/build.py]: disagree. Locally we don't want to use cache, this is just an artifact for CI. please merge this PR.

Comment (marcoabreu) [reply ci/build.py]: No, it is intended to be used locally as well.

**Pull request merged**